server/src/main/java/org/apache/druid/curator/discovery/CuratorDruidLeaderSelector.java

```diff
@@ -208,28 +209,26 @@ public void unregisterListener()
     listenerExecutor.shutdownNow();
   }
 
-  private void stopAndCreateNewLeaderLatch()
+  private void closeLeaderLatchQuietly()
```

Comment:
This still does close and re-create the latch, how about naming it `closeAndRecreateLeaderLatchQuietly`? The name `closeLeaderLatchQuietly` loses the sense that it's also creating a new latch.

Reply:
Done.